source/opt/ir_context.h

```diff
@@ -217,16 +217,22 @@ class IRContext {
   // Invalidates the analyses marked in |analyses_to_invalidate|.
   void InvalidateAnalyses(Analysis analyses_to_invalidate);
 
-  // Turns the instruction defining the given |id| into a Nop. Returns true on
+  // Deletes the instruction defining the given |id|. Returns true on
   // success, false if the given |id| is not defined at all. This method also
   // erases both the uses of |id| and the information of this |id|-generating
   // instruction's uses of its operands.
+  //
+  // Pointer and iterator pointing to the deleted instruction become invalid.
+  // However other pointer and iterator are still valid.
   bool KillDef(uint32_t id);
 
-  // Turns the given instruction |inst| to a Nop. This method erases the
+  // Deletes the given instruction |inst|. This method erases the
   // information of the given instruction's uses of its operands. If |inst|
   // defines an result id, the uses of the result id will also be erased.
-  void KillInst(ir::Instruction* inst);
+  //
+  // Pointer and iterator pointing to the deleted instruction become invalid.
+  // However other pointer and iterator are still valid.
```

Comment:
Comment on behaviour of special instructions like OpLabel and OpFunction.

Reply:
Fixed up the the comments for KillInst and KillDef in general.